clang/test/SemaSYCL/reqd-work-group-size-device.cpp

```diff
@@ -6,19 +6,49 @@
 using namespace cl::sycl;
 queue q;
 
-[[cl::reqd_work_group_size(4, 1, 1)]] void f4x1x1() {} // expected-note {{conflicting attribute is here}}
+[[sycl::reqd_work_group_size(4, 1, 1)]] void f4x1x1() {} // expected-note {{conflicting attribute is here}}
 // expected-note@-1 {{conflicting attribute is here}}
-[[cl::reqd_work_group_size(32, 1, 1)]] void f32x1x1() {} // expected-note {{conflicting attribute is here}}
+[[sycl::reqd_work_group_size(32, 1, 1)]] void f32x1x1() {} // expected-note {{conflicting attribute is here}}
 
-[[cl::reqd_work_group_size(16, 1, 1)]] void f16x1x1() {}   // expected-note {{conflicting attribute is here}}
-[[cl::reqd_work_group_size(16, 16, 1)]] void f16x16x1() {} // expected-note {{conflicting attribute is here}}
+[[sycl::reqd_work_group_size(16, 1, 1)]] void f16x1x1() {} // expected-note {{conflicting attribute is here}}
+[[sycl::reqd_work_group_size(16, 16, 1)]] void f16x16x1() {} // expected-note {{conflicting attribute is here}}
 
-[[cl::reqd_work_group_size(32, 32, 1)]] void f32x32x1() {}   // expected-note {{conflicting attribute is here}}
-[[cl::reqd_work_group_size(32, 32, 32)]] void f32x32x32() {} // expected-note {{conflicting attribute is here}}
+[[sycl::reqd_work_group_size(32, 32, 1)]] void f32x32x1() {} // expected-note {{conflicting attribute is here}}
+[[sycl::reqd_work_group_size(32, 32, 32)]] void f32x32x32() {} // expected-note {{conflicting attribute is here}}
+
+// No diagnostic because the attributes are synonyms with identical behavior.
+[[intel::reqd_work_group_size(4, 4, 4)]] void four();
+[[sycl::reqd_work_group_size(4, 4, 4)]] void four(); // OK
+
+// Same for the default values.
+[[intel::reqd_work_group_size(4)]] void four_again();
+[[sycl::reqd_work_group_size(4)]] void four_again(); // OK
+[[intel::reqd_work_group_size(4, 1, 1)]] void four_again(); // OK
+[[sycl::reqd_work_group_size(4, 1, 1)]] void four_again(); // OK
```

Comment:
Could you change one of these to '(4, 1)' just so that variant is also checked?

Reply:
Yes, but there's a new, exciting bug here that needs to be fixed (not a bug caused by my changes, just a misunderstanding of the spec that these tests highlight).